Source/WTF/wtf/AccessibleAddress.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include <wtf/AccessibleAddress.h>
+
+#include <mutex>
+#include <wtf/MathExtras.h>
+#include <wtf/PageBlock.h>
+#include <wtf/StdLibExtras.h>
+
+#if OS(DARWIN)
+#include <dlfcn.h>
+#include <mach-o/getsect.h>
+#include <mach-o/ldsyms.h>
+#include <mach/vm_param.h>
+#endif
+
+namespace WTF {
+
+uintptr_t lowestAccessibleAddress()
+{
+    static uintptr_t cached = 0;
+    static std::once_flag onceKey;
+    std::call_once(onceKey, []() {
```

Comment:
Instead of using a `static` field and doing a `once_flag` check, can you put this field in `WTF::Config` and create an `initializeAccessibleAddress` to be called from `WTF::initialize` (in Threading.cpp) instead?  That way, `lowestAccessibleAddress()` can be an inline function.  `lowestAccessibleAddress()` can also ASSERT that the field in `WTF::Config` is initialized (i.e. not null).  I think we want `isSanePointer` to be cheap.

Maybe we should make `highestAccessibleAddress` an inline function too.

Reply:
That's nice! Changed.